oauth2_provider/views/base.py

```diff
@@ -87,6 +88,14 @@ def get_initial(self):
         }
         return initial_data
 
+    def get_application(self, client_id):
+        """
+        get an application from a client_id, this caches the applications
+        """
+        if client_id not in self._application_cache:
```

Comment:
I don't think that's better. But that aside this cache can be pulled into its own PR.